packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx

```diff
@@ -102,26 +102,28 @@ export const Result: React.FunctionComponent<ResultProps> = ({
     fetchDocumentsResponseStore: { isLoading }
   } = useContext(SearchContext);
 
-  const firstPassage: DiscoveryV2.QueryResultPassage | undefined = get(
-    result,
-    'document_passages[0]'
-  );
-  const firstPassageText = get(firstPassage, 'passage_text');
-  const hasPassage = usePassages && !!firstPassageText;
-  let displayedText: string | undefined = get(result, bodyField);
-  if (hasPassage) {
-    displayedText = firstPassageText;
+  const passages: DiscoveryV2.QueryResultPassage[] =
+    result?.document_passages?.filter(passage => !!passage?.passage_text) || [];
+  const passageTexts: string[] = passages?.map(passage => passage.passage_text!) || [];
+  const hasPassages = usePassages && !!passageTexts && passageTexts.length > 0;
+  let displayedTexts: string[] | undefined = get(result, bodyField)
+    ? [get(result, bodyField)]
+    : undefined;
```

Comment:
You can still replace this `get()` with optional chaining:
```suggestion
  let displayedTexts: string[] | undefined = result?.[bodyField]
    ? [result?.[bodyField]]
    : undefined;
```

Reply:
I was hitting issues where the values being sent in were of the form `abc.def`-- and I wasn't sure how to handle that with optional chaining. I.e. `result?.[bodyField]` was coming back `undefined` in those cases (since it was looking for a single field called `abc.def`, rather than two layers of fields).

Reply:
Ah, that's right. Forgot that `bodyField` can have multiple levels. Ignore my suggestion, then.